### PR TITLE
Add wordpress information

### PR DESCRIPTION
**Pull request by lilyinstarlight**

### CODE REVIEW

File: services/wordpress/wordpress.tex [hunk 1]
```diff
@@ -0,0 +1,26 @@
+\section{Wordpress}\index{wordpress}
+
+Wordpress is a PHP content management system. It has reasonable security in a new default install, but has a poor track record for remote execution exploits. The best way to secure Wordpress is to update it if possible and remove all unnecessary or old plugins. The following will help limit damage from possible remaining exploits:
+
+\begin{itemize}
+	\item make sure file permissions are good
```
Comment (robertu94) [inline services/wordpress/wordpress.tex]: Needs to be more explicit.  Read only, Read/Write for which users/groups

File: services/wordpress/wordpress.tex [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	\begin{itemize}
+		\item `/` needs to be writable only by user account
```
Comment (robertu94) [inline services/wordpress/wordpress.tex]: Which user

File: services/wordpress/wordpress.tex [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+		\item `/wp-content` needs to be writable by web server
+		\item `/wp-content/plugins` needs to be writable only by user account
+	\end{itemize}
+
+	\item remove unnecessary database permissions
+	\begin{itemize}
+		\item only allow the SQL user to have \lstinline|SELECT,INSERT,UPDATE,DELETE|
```
Comment (robertu94) [inline services/wordpress/wordpress.tex]: On what tables?

File: services/wordpress/wordpress.tex [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	\end{itemize}
+
+	\item disable file editing from wp-admin
+	\begin{itemize}
+		\item add \lstinline|define('DISALLOW_FILE_EDIT', true);| to \lstinline|wp-config.php|
+	\end{itemize}
+
+	\item move \lstinline|wp-config.php| to the directory above the wordpress root
+
+	\item add 'AskApache Password Protect'
```
Comment (robertu94) [inline services/wordpress/wordpress.tex]: Are there additional settings to set on Apache that need to go in with this change?

File: services/wordpress/wordpress.tex [hunk 5]
```diff
@@ -0,0 +1,26 @@
+\section{Wordpress}\index{wordpress}
+
+Wordpress is a PHP content management system. It has reasonable security in a new default install, but has a poor track record for remote execution exploits. The best way to secure Wordpress is to update it if possible and remove all unnecessary or old plugins. The following will help limit damage from possible remaining exploits:
```
Comment (robertu94) [inline services/wordpress/wordpress.tex]: Perhaps list where to get source files/install them if it is not in the software repository for the distribution that you are using.

File: services/wordpress/wordpress.tex (hunk 4 above)
Comment (lilyinstarlight) [inline services/wordpress/wordpress.tex]: Only .htaccess has to be enabled. I'm going to make it more specific, but AskApache Password Protect basically adds HTTP authentication to the admin panel preventing bugs there from being exploited.

File: services/wordpress/wordpress.tex (hunk 3 above)
Comment (lilyinstarlight) [inline services/wordpress/wordpress.tex]: All of the wp_\* tables. I'll add that here and I'll add a full MySQL/MariaDB query to (probably) fix the permissions for you.

File: services/wordpress/wordpress.tex (hunk 2 above)
Comment (lilyinstarlight) [inline services/wordpress/wordpress.tex]: The user will probably be 'www', but it is whoever owns the directory and manages www files.

File: services/wordpress/wordpress.tex (hunk 5 above)
Comment (lilyinstarlight) [inline services/wordpress/wordpress.tex]: The download is https://wordpress.org/latest.tar.gz and I'll add a few stuff (basically SQL commands) for the setup.

File: services/wordpress/wordpress.tex (hunk 1 above)
Comment (lilyinstarlight) [inline services/wordpress/wordpress.tex]: This is specified by the nested itemize below it. If you would like for me to format that differently, I'll take suggestions.

Comment (robertu94) [inline services/wordpress/wordpress.tex]: I just did a single pass, I think once you have added the other things that you said that you will add, we should be good.

File: services/wordpress/sql/wordpress.sql [hunk 6]
```diff
@@ -0,0 +1,4 @@
+CREATE USER wordpress@localhost IDENTIFIED BY 'password';
+CREATE DATABASE wordpress;
+GRANT GRANT SELECT,INSERT,UPDATE,DELETE ON wordpress.* TO wordpress@localhost;
```
Comment (robertu94) [inline services/wordpress/sql/wordpress.sql]: Are the two "grants" a typo?

File: services/wordpress/sql/wordpress.sql [hunk 7]
```diff
@@ -0,0 +1,4 @@
+CREATE USER wordpress@localhost IDENTIFIED BY 'password';
```
Comment (robertu94) [inline services/wordpress/sql/wordpress.sql:1]: Won't this store the password in plain text?

Comment (lilyinstarlight) [inline services/wordpress/sql/wordpress.sql:1]: No, it shouldn't unless MySQL is misconfigured.

File: services/wordpress/sql/wordpress.sql (hunk 6 above)
Comment (lilyinstarlight) [inline services/wordpress/sql/wordpress.sql]: It is indeed.

**Comment by robertu94** [comment]
Ok, sounds good.

**Pull request merged**